geth/node/manager.go

```diff
@@ -563,3 +565,16 @@ func (m *NodeManager) RPCServer() (*rpc.Server, error) {
 
 	return m.rpcServer, nil
 }
+
+// isNodeAvailable check if we have a node running and make sure is fully started
+func (m *NodeManager) isNodeAvailable() error {
+	if m.node == nil {
```

Comment:
Since the `node` field is supposed to be secured through the mutex, this should call:

```
defer m.Unlock()
m.Lock()
```

Before it's contents.

Reply:
I agree, it would be the safest to add it here. But `NodeManager`'s methods need a closer look anyway as some of them don't use the mutex without a reason.